capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -121,34 +120,35 @@ public WeeklyPageView getCurrentPageViews(String restaurantName, String restaura
         // Query with double sort so the first element of every restaurant should be
         // the earliest week and year.
         Query query = new Query("PageViews")
-                .addSort("week", Query.SortDirection.ASCENDING)
-                .addSort("year", Query.SortDirection.ASCENDING);
+                .addSort("year", Query.SortDirection.ASCENDING)
+                .addSort("week", Query.SortDirection.ASCENDING);
         PreparedQuery preparedQuery = dataStore.prepare(query);
         List<Entity> results = preparedQuery.asList(FetchOptions.Builder.withDefaults());
 
-        // Map to maintain a reference to each restaurant and make searching for them easy.
+        // Map to maintain a reference to each restaurantList and make searching for them easy.
         // The map and the list share the reference to each restaurantPageViews.
-        Map<String, RestaurantPageViews> restaurantNameMap = new HashMap<>();
-        List<RestaurantPageViews> restaurantPageViewsList = new ArrayList<>();
-        RestaurantPageViews currRestaurant;
+        Map<String, List<WeeklyPageView>> restaurantNameMap = new HashMap<>();
 
         for (Entity entity : results) {
             String name = entity.getProperty("restaurantName").toString();
             int year = Integer.parseInt(entity.getProperty("year").toString());
```

Comment:
How is `int` modeled internally, when retrieving an entity? It's weird to me that you would need to convert it to a string and then parse the string as an int. Usually, these generic value types have a `intValue()` method or somesuch, which will return the right thing without needing to parse, and which will throw an exception if your assumption about what it is is incorrect.

Reply:
So they always come back as a Entity Object, and they don't seem to have a method to change it specifically to an int straight away. You have to do the toString() first and then change it.